### PR TITLE
Returning nearest sites based on latitude and longitude

### PR DESCRIPTION
**Pull request by NoahNsimbe**
# Feature

**_WHAT DOES THIS PR DO?_**
Returns the nearest sites that are within the specified radius (km) of the provided latitude and longitude

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
None

**_WHAT IS THE LINK TO THE PR BRANCH_**
[nearest-site](https://github.com/airqo-platform/AirQo-api/tree/nearest-site)

**_HOW DO I TEST OUT THIS PR?_**
```bash
cd AirQo-api/src/device-registry
npm install
npm run stage-mac
```
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
```http
http://localhost:3000/api/v1/devices/sites/nearest?tenant=airqo&latitude=1.714471&longitude=33.621236&radius=1
```

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
None

**_ARE THERE ANY RELATED PRs?_**
None

### CODE REVIEW

File: src/device-registry/routes/api-v1.js [hunk 1]
```diff
@@ -1361,6 +1361,7 @@ router.delete(
   siteController.delete
 );
 router.post("/sites/nearest", siteController.findNearestSite);
+router.post("/sites/nearestSiteByCoordinates", siteController.findNearestSiteByCoordinates);
```
Comment (Baalmart) [inline src/device-registry/routes/api-v1.js:1364]: Could we transfer this logic to "/sites/nearest" route?

File: src/device-registry/utils/create-site.js [hunk 2]
```diff
@@ -852,6 +853,22 @@ const manageSite = {
       logElement("server error", e.message);
     }
   },
+
+  findNearestSitesByCoordinates: (sites, radius, latitude, longitude ) => {
+    let nearest_sites = []
```
Comment (Baalmart) [inline src/device-registry/utils/create-site.js:858]: Thanks @NoahNsimbe ,could we do some error handling for this function?

File: src/device-registry/utils/distance.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const distanceBtnTwoPoints = (latitude1, longitude1, latitude2, longitude2) => {
+
```
Comment (Baalmart) [inline src/device-registry/utils/distance.js:2]: Please add error handling to this function @NoahNsimbe

File: src/device-registry/controllers/create-site.js [hunk 4]
```diff
@@ -269,6 +269,47 @@ const manageSite = {
     }
   },
 
+  findNearestSiteByCoordinates: async (req, res) => {
+    try {
```
Comment (Baalmart) [inline src/device-registry/controllers/create-site.js:273]: Thanks @NoahNsimbe , glad we are handling errors here, awesome stuff!
Could you double check the relevance of the double "try catch statements" here?

**Review by Baalmart — CHANGES_REQUESTED** [review]
Thanks @NoahNsimbe , this is awesome!
Just left a few comments for your review. Shall run the PR in due course to test out the functionality.

**Review by danielogen — APPROVED** [review]
LGTM

**Pull request merged**